### PR TITLE
Unit change signals

### PR DESCRIPTION
**Pull request by phlax**
move unit mangling to unit pre_save handler

### CODE REVIEW

File: pootle/apps/pootle_store/receivers.py [hunk 1]
```diff
@@ -28,6 +34,60 @@ def handle_suggestion_added(**kwargs):
...
+        unit.submitted_on = None
+
+    # Updating unit from the .po file set its revision property to
+    # a new value (the same for all units during its store updated)
+    # since that change doesn't require further sync but note that
+    # auto_translated units require further sync
+    update_revision = (
+        not unit.revision
+        or (not unit.revision_updated
+            and (unit.updated and not auto_translated)))
```
Comment (unho) [inline pootle/apps/pootle_store/receivers.py]: I would invert the order and check `not auto_translated` first.

Comment (phlax) [inline pootle/apps/pootle_store/receivers.py]: yep, k

File: pootle/apps/pootle_store/models.py [hunk 2]
```diff
@@ -263,69 +276,15 @@ def save(self, *args, **kwargs):
         changed_with = kwargs.pop("changed_with", None) or SubmissionTypes.SYSTEM
         reviewed_by = kwargs.pop("reviewed_by", None)
         reviewed_on = kwargs.pop("reviewed_on", None)
-        revision = kwargs.pop('revision', None)
         submitted_by = kwargs.pop("submitted_by", None)
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: I think this line needs changing too:

```python
        submitted_by = kwargs.pop("submitted_by", self.submitted_by)
```

File: pootle/apps/pootle_store/receivers.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/receivers.py]: actually not - i think its more confusing having the neg at the front

File: pootle/apps/pootle_store/models.py [hunk 3]
```diff
@@ -278,64 +278,13 @@ def save(self, *args, **kwargs):
         reviewed_on = kwargs.pop("reviewed_on", None)
         submitted_by = kwargs.pop("submitted_by", None)
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: Do not forget to default to `self.submitted_by` here.

Comment (phlax) [inline pootle/apps/pootle_store/models.py]: i would rather not as that field will be removed asap

Comment (phlax) [inline pootle/apps/pootle_store/models.py]: ..and doesnt seem to break anything

**Pull request merged**